plugin.js

```diff
@@ -106,3 +113,67 @@ function addToPath(tree, p, value) {
   p.shift();
   addToPath(child, p, value);
 }
+
+function getBytesPerFileUsingSourceMap(rendered) {
+
+  let map = new SourceMapConsumer(rendered.map);
+  let lines = rendered.code.split(/[\r\n]/);
+
+  let bytesPerFile = {};
+
+  // For every byte in the minified code, do a sourcemap lookup.
+  for (let line = 0; line < lines.length; line++) {
+    for (let col = 0; col < lines[line].length; col++) {
+      let result = map.originalPositionFor({ line: line + 1, column: col });
+      let source = result.source || 'root'
+      if (!bytesPerFile[source]) bytesPerFile[source] = 0;
+      bytesPerFile[source]++
+    }
+  }
+  return Object.keys(bytesPerFile).map(file =>
+      ({ file: path.resolve(file), bytes: bytesPerFile[file] }));
+} 
+
+
+
+// Given a file /path/to/file/on/filesystem.js
+// - remove extension
+// - return `segmentCount` number of path segments, starting from the tail and working backwards
+// segments('/path/to/file/on/filesystem.js')(3) === ['filesystem', 'on', 'file']
+const segments = filepath => segmentCount =>
+  filepath.replace(/\.\w{1,6}$/, "").split(path.sep).reverse().slice(0, segmentCount);
```

Comment:
Why does it need to remove extension? I beleave rollup will rest original extension in module and will use the same in source map, otherwise this would be kind of silly